src/filebuf.h

```diff
@@ -19,7 +19,7 @@
 
 #define GIT_FILEBUF_HASH_CONTENTS		(1 << 0)
 #define GIT_FILEBUF_APPEND				(1 << 2)
-#define GIT_FILEBUF_FORCE				(1 << 3)
+#define GIT_FILEBUF_CREATE_MISS_DIR		(1 << 3)
```

Comment:
This is not only a rename of the flag, but in fact you completely change semantics. Previously, this flag would've just removed pre-existing locks, while the new flag will just create leading directories. I take it that the `GIT_FILEBUF_FORCE` flag is not really needed anymore after introducing `GIT_FILEBUF_CREATE_MISS_DIR`, in which case it's fine to remove it. But when one goes back in history at a later point in time, it's really confusing when one reads about "renaming" a flag when what you're really doing is creating a new one and removing the old now-unused one in the same step.

I'd also vote for a flag name "GIT_FILEBUF_CREATE_LEADING_DIRS"

Reply:
I agree for your flag name. But I don't really think that this is a new flag. I think the intention and usage of the flag was exactly to what your suggested name describes. The unlink was a bug due to the bad naming. Actually I don't really understand how it could get introduced, as it was even the inventor of the flag who also added the unlink. What do you propose as a resolution? I think a good explanation commit message should be ok.